sdks/python/src/opik/evaluation/metrics/conversation/session_completeness/metric.py

```diff
@@ -0,0 +1,282 @@
+    intentions from the conversation turns. The same LLM is then used to assess
+    whether each intention was addressed and/or fulfilled over the course of
+    the conversation. It returns a score between 0.0 and 1.0, where higher values
+    indicate better session completeness.
+
+    Args:
+        model: The language model to use for evaluation.
+        name: The name of the metric. The default is "session_completeness_quality".
+        include_reason: Whether to include a reason for the score.
+        track: Whether to track the metric. Default is True.
+        project_name: The project name to track the metric in.
+
+    Example:
+        >>> from opik.evaluation.metrics import SessionCompletenessQuality
+        >>> conversation = [
+        >>>     {"role": "user", "content": "Hello!"},
+        >>>     {"role": "assistant", "content": "Hi there!"},
+        >>>     {"role": "user", "content": "How are you?"},
+        >>>     {"role": "assistant", "content": "I'm doing well, thank you!"},
+        >>> ]
+        >>> metric = SessionCompletenessQuality()
+        >>> result = metric.score(conversation)
+        >>> if result.scoring_failed:
+        >>>     print(f"Scoring failed: {result.reason}")
+        >>> else:
+        >>>     print(result.value)
+        0.95
+    """
+
+    def __init__(
+        self,
+        model: Optional[Union[str, base_model.OpikBaseModel]] = None,
+        name: str = "session_completeness_quality",
+        include_reason: bool = True,
+        track: bool = True,
+        project_name: Optional[str] = None,
+    ):
+        super().__init__(
+            name=name,
+            track=track,
+            project_name=project_name,
+        )
+        self._init_model(model)
+        self._include_reason = include_reason
+
+    def _init_model(
+        self, model: Optional[Union[str, base_model.OpikBaseModel]]
+    ) -> None:
+        if isinstance(model, base_model.OpikBaseModel):
+            self._model = model
+        else:
+            self._model = models_factory.get(model_name=model)
+
+    def score(
+        self,
+        conversation: List[Dict[str, str]],
+        **ignored_kwargs: Any,
+    ) -> score_result.ScoreResult:
+        return self._calculate_score(conversation=conversation)
+
+    async def ascore(
+        self,
+        conversation: List[Dict[str, str]],
+        **ignored_kwargs: Any,
+    ) -> score_result.ScoreResult:
+        return await self._a_calculate_score(conversation=conversation)
+
+    def _extract_user_goals(
+        self,
+        conversation: List[Dict[str, str]],
+    ) -> List[str]:
+        llm_query = templates.extract_user_goals(conversation)
+        model_output = self._model.generate_string(
+            input=llm_query, response_format=schema.UserGoalsResponse
+        )
+        try:
+            return schema.UserGoalsResponse.model_validate_json(model_output).user_goals
```

Comment:
Sometimes users use models that don't support response_format and might send something like 'Here you go: {"key": "value", ...}'
As a last line of defense in such cases we use the logic from `llm_judges.parsing_helpers` (search for "{" and "}" and try to parse what's in the middle :))